core/libraries/batch/JobHandlers/PreviewEventDeletion.php

```diff
@@ -112,6 +152,54 @@ protected function createModelObjNodes($model_objs, array $dont_traverse_models
         return $nodes;
     }
 
+    /**
+     * Gets all the transactions related to these events that aren't related to other events. They'll be deleted too.
+     * (Ones that are related to other events can stay around until those other events are deleted too.)
+     * @since $VID:$
+     * @param $event_ids
+     * @return array of transaction IDs
+     */
+    protected function getTransactionsToDelete($event_ids)
+    {
+        if (empty($event_ids)) {
+            return [];
+        }
+        global $wpdb;
+        $event_ids = array_map(
+            'intval',
+            $event_ids
+        );
+        $imploded_sanitized_event_ids = implode(',', $event_ids);
+        // Select transactions with registrations for the events $event_ids which also don't have registrations
+        // for any events NOT in $event_ids.
+        // Notice the outer query searched for transactions whose registrations ARE in $event_ids,
+        // whereas the inner query checks if the outer query's transaction has any registrations that are
+        // NOT IN $event_ids (ie, don't have registrations for events we're not just about to delete.)
```

Comment:
I hope the above explanation will help down the line...

Reply:
oh ya that's great thnx... 
I just had a brain fart moment yesterday where I initially didn't see the `NOT` and then my brain just refused to see it from that point on... or maybe I was just focusing too much on other aspects of the query.